<?xml version="1.0"?>
<launch>
  <include file="$(find tsim)/launch/turtle_pent.launch" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find nuturtle_description)/config/config.rviz" />

  <node pkg="rigid2d" type="odometer" name="odometer" output="screen">
    <param name="joint_state_topic" value="/joint_states"/>
    <param name="odom_frame_id" value="/odom_frame_id"/>
    <param name="base_link" value="/base_link"/>
  </node>

  <param name="robot_description"
       command="$(find xacro)/xacro $(find nuturtle_description)/urdf/diff_drive.urdf.xacro"/>
  <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />

  <node pkg="rigid2d" type="fake_diff_encoders" name="fake_diff_encoders"  output="screen">
    <param name="joint_state_topic" value="/joint_states"/>
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  <!-- <node name="set_pose_service" pkg="rigid2d" type="set_pose_service"/> -->

</launch>
